nexus/completion.go

```diff
@@ -69,8 +80,10 @@ func NewOperationCompletionSuccessful(result any, options OperationCompletionSuc
 		header := http.Header{"Content-Length": []string{strconv.Itoa(len(content.Data))}}
 
 		return &OperationCompletionSuccessful{
-			Header: addContentHeaderToHTTPHeader(content.Header, header),
-			Body:   bytes.NewReader(content.Data),
+			Header:     addContentHeaderToHTTPHeader(content.Header, header),
```

Comment:
We've typically let user provided headers override any well known headers that the SDK sets. Let's preserve this behavior here.

Reply:
I updated the logic in the `applyToHTTPRequest` methods. LMK if that looks good to you.